File: src/events/component_interaction/instructions.rs
use poise::serenity_prelude as serenity;

use serenity::*;

use crate::prelude::*;

pub async fn continue_clicked(
    context: &Context,
    component_interaction: &MessageComponentInteraction,
    bot_data: &BotData,
) -> BotResult<()> {
    // Update the user's registration state to started
    sqlx::query!(
        "update registrations
        set status = ?
        where user_id = ?",
        RegistrationStatus::Started.to_string(),
        component_interaction.user.id.0,
    )
    .execute(&bot_data.database_pool)
    .await?;

    // Ask the user to enter their name
    component_interaction
        .create_followup_message(&context.http, |message| {
            message.embed(embeds::registration::name_request())
        })
        .await?;

    // Log the start of registration
    bot_data
        .channels
        .log_channel_id
        .send_message(&context.http, |message| {
            message.embed(embeds::logs::registration_started(
                component_interaction.user.id,
            ))
        })
        .await?;

    Ok(())
}

File: norris/events/component_interaction/instructions.py
from discord import Interaction
from sqlalchemy import update

from ...bot import Norris
from ...model import Registration, RegistrationStatus
from . import verify_registration_status


async def continue_clicked(interaction: Interaction, norris: Norris) -> None:
    # Defer response to give time for database queries
    await interaction.response.defer()

    # Check that the user has the correct state to click the button
    if not await verify_registration_status(interaction.user.id,
                                            RegistrationStatus.UNREGISTERED,
                                            norris):
        return

    async with norris.database_engine.begin() as connection:
        # Update the user's registration state to started
        await connection.execute(
            update(Registration)
            .values(status=RegistrationStatus.STARTED)
            .where(Registration.user_id == interaction.user.id),
        )

    # NOTE: I love circular imports, what an amazing module system Python has
    from ...responses import embeds

    # Ask the user to enter their name
    await interaction.followup.send(embed=embeds.registration.request_name_embed())

    # Log the start of registration
    await norris.get_channel(norris.channels.log_channel_id).send(
        embed=embeds.logs.registration_started_log_embed(interaction.user.id),
    )
